src/ucp/core/ucp_am.c

```diff
@@ -1254,13 +1300,26 @@ ucs_status_t ucp_am_rndv_process_rts(void *arg, void *data, size_t length,
                                           rts->super.sreq.ep_id);
     status          = am_cb->cb(am_cb->context, hdr, rts->am.header_length,
                                 desc + 1, rts->super.size, &param);
-    if (status != UCS_INPROGRESS) {
-        /* TODO: Check that recv was not called and if not, send
-         * reject to the peer. */
-        return UCS_OK;
+    if ((status == UCS_INPROGRESS) ||
+        (desc->flags & UCP_RECV_DESC_FLAG_RNDV_STARTED)) {
+        /* User either wants to save descriptor for later use or initiated
+         * rendezvous receive (by ucp_am_recv_data_nbx) in the callback. */
+        ucs_assert(!UCS_STATUS_IS_ERR(status));
+        return desc_status;
+    }
+
```

Comment:
maybe add comment: user did not want to receive the data, fall through to sending ATS